xdsl/ir/context.py

```diff
@@ -0,0 +1,533 @@
+from collections.abc import Callable, Iterable
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from xdsl.ir import Attribute, Dialect, Operation
+
+
+def get_all_dialects() -> "dict[str, Callable[[], Dialect]]":
```

Comment:
Why do you need the "" here? Can you `from __future__ import annotations`? I forgot if this is supposed to work or not.

Reply:
I'm not sure why annotations? quoted type hints are for not adding runtime dependencies because of type hints. There is a cyclic dependency if I don't use quotes here; Should be solved by moving this function to its own file!

Reply:
Indeed, solved